packages/react-dom/src/events/__tests__/ChangeEventPlugin-test.js

```diff
@@ -12,57 +12,72 @@
 var React = require('react');
 var ReactDOM = require('react-dom');
 var ReactTestUtils = require('react-dom/test-utils');
-// TODO: can we express this test with only public API?
-var ChangeEventPlugin = require('../ChangeEventPlugin').default;
-var inputValueTracking = require('../../client/inputValueTracking');
 
 function getTrackedValue(elem) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  return tracker.getValue();
+  return elem.value;
 }
 
-function setTrackedValue(elem, value) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  tracker.setValue(value);
-}
 
-function setUntrackedValue(elem, value) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  var current = tracker.getValue();
 
-  if (elem.type === 'checkbox' || elem.type === 'radio') {
-    elem.checked = value;
-  } else {
-    elem.value = value;
-  }
-  tracker.setValue(current);
+var setUntrackedChecked = Object.getOwnPropertyDescriptor(
+  HTMLInputElement.prototype,
+  'checked',
+).set;
+
+var setUntrackedValue = Object.getOwnPropertyDescriptor(
+  HTMLInputElement.prototype,
+  'value',
+).set;
+
+function getEvent(type, opts) {
+  var event = document.createEvent('Event');
+  event.initEvent(type, true, true);
+  return event;
+}
+
+function dispatchEventOnNode(node, type) {
+  //console.log(`Dispatch ${type}`, node.checked);
+  var res = node.dispatchEvent(
+    new Event( type, {bubbles: true, cancelable: true} ));
 }
 
 describe('ChangeEventPlugin', () => {
-  it('should fire change for checkbox input', () => {
+  it('should fire click for checkbox input', () => {
     var called = 0;
 
     function cb(e) {
       called = 1;
-      expect(e.type).toBe('change');
+      expect(e.type).toBe('click');
     }
 
-    var input = ReactTestUtils.renderIntoDocument(
-      <input type="checkbox" onChange={cb} />,
+    var container = document.createElement('div');
+
+    var stub = ReactDOM.render(
+      <input type="checkbox" onClick={cb}/>,
+      container
     );
 
-    setUntrackedValue(input, true);
-    ReactTestUtils.SimulateNative.click(input);
+    document.body.appendChild(container);
 
-    expect(called).toBe(1);
+    var node = ReactDOM.findDOMNode(stub);
```

Comment:
Tip: there is no "stub" :-) Old tests are bad, don't copy them.
`ReactDOM.render()` already returns a DOM node.